packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1507,7 +1520,7 @@ export class ContainerRuntime extends EventEmitter
                     sequenceNumber: message.sequenceNumber,
                     snapshot: attachMessage.snapshot ?? {
                         id: null,
-                        entries: [createAttributesBlob(pkg)],
+                        entries: [createAttributesBlob(pkg, true /* isRootDataStore */)],
```

Comment:
This does not look right - Why is it true, not the real value for isRoot from summary?
@arinwt, any chance you can clarify what this code is doing? I.e. why we have only attributes blob here, not full summary we got from attach op? 
I think this creates a base for "safe" summary when summarization fail. If so, why attributes is enough?

Reply:
I believe this is for backwards compat which handles the case where an attach message does not have snapshot. I am setting this to true because in this case, we should treat the data store as root otherwise we might end up collecting data store which is in old format. Right?

Reply:
@arinwt, can you please take a look? What is the purpose of this snapshot? I'd think it has to be complete snapshot - whatever came on the wire.

Reply:
Navin is correct. The typing for `InboundAttachMessage` was created specifically for this back-compat handling. See my comment [here](https://github.com/microsoft/FluidFramework/blob/daa2802e05cadb452da1bfcc1e906a908c471464/packages/runtime/runtime-definitions/src/protocol.ts#L72) for more information.

Since summarizing cleans these up, only replaying old ops or docs that haven't been opened in a long time would cause errors here, so we could stop handling this back-compat case, but we should be deliberate about it and remove this place as well.

Reply:
I was thinking about this, and also the discussion in the other thread where we were talking about how to get this information (along with the cached routes ) for GC without realizing the data store.
From one of Vlad's suggestion and my discussion with Arin, it makes sense to add "isRootDataStore" and the cached routes into a separate blob. So, we can only read this blob from the snapshot if we don't need to realize the data store.
Adding this to the attributes blob means we will have to read the attributes blob too. This should be fine but adding the cached routes to attributes doesn't make much sense and so we need an additional blob anyway.
It makes the code cleaner and well encapsulated. I will update this PR to do that.

Reply:
Can we delete this handling (maybe in separate PR)? There is no need to write out files in such old format - only reading is required.

Reply:
Do you mean remove the garbage collection attributes in the data store snapshot in old format? In `snapshot` and `summarizeInternal` apis?

Reply:
No, remove ability to write out attach op in old format (without snapshot) - I think that's what this snapshot is about.

Reply:
This use-case is within scope of _receiving_ an attach op. If the attach op we read is missing snapshot, then we create a basic one to fix it going forward.

Removing this handling here would be equivalent to entire summary failure in the intersection of two problems: 
1. outstanding attach op without snapshot (really old doc)
2. first summary fails for that data store (may be somewhat likely for really old doc, but not sure)